scripts/codoon_sync.py

```diff
@@ -75,6 +84,139 @@ def download_codoon_gpx(gpx_data, log_id):
+    # Activities
+    activities = ET.Element("Activities")
+    training_center_database.append(activities)
+    # sport type
+    sports_type = TYPE_DICT.get(run_data["sports_type"])
+    # activity
+    activity = ET.Element("Activity", {"Sport":sports_type})
+    activities.append(activity)
+    #   Id
+    activity_id = ET.Element("Id")
+    activity_id.text = fit_id
+    activity.append(activity_id)
+    #   Creator
+    activity_creator = ET.Element("Creator")
+    activity.append(activity_creator)
+    #       Name
+    activity_creator_name = ET.Element("Name")
+    activity_creator_name.text = "咕咚"
+    activity_creator.append(activity_creator_name)
+    #   Lap
+    fit_start_time_CH = run_data["start_time"]
+        # parse to time array
+    time_array = time.strptime(fit_start_time_CH, "%Y-%m-%dT%H:%M:%S")
+        # parse to unix timestamp
+    unix_time = int(time.mktime(time_array))
+    unix_time = unix_time - 28800   # move to UTC[for UTC+08:00 only]
+        # zulu time
+    fit_start_time = time.strftime("%Y-%m-%dT%H:%M:%SZ",time.localtime(unix_time))
+    activity_lap = ET.Element("Lap",{"StartTime":fit_start_time})
+    activity.append(activity_lap)
+    #       TotalTimeSeconds
+    activity_lap.append(formated_input(run_data,"total_time","TotalTimeSeconds"))
+    #       DistanceMeters
+    activity_lap.append(formated_input(run_data,"total_length","DistanceMeters"))
+    #       Calories
+    activity_lap.append(formated_input(run_data,"total_calories","Calories"))
+    
+    
+    # Track
+    track = ET.Element("Track")
+    activity_lap.append(track)
+    for i in fit_array:
+        tp = ET.Element("Trackpoint")
+        track.append(tp)
+        # Time
+        time_stamp = time.strftime("%Y-%m-%dT%H:%M:%SZ",time.localtime(i["time"]))
+        time_label = ET.Element("Time")
+        time_label.text = time_stamp
+        tp.append(time_label)
+        # HeartRateBpm
+            # None was converted to bytes by np.dtype, becoming a string "None" after decode...-_-
+            # as well as LatitudeDegrees and LongitudeDegrees below
+        if not bytes.decode(i["bpm"]) == "None":
+            bpm = ET.Element("HeartRateBpm")
+            bpm_value = ET.Element("Value")
+            bpm.append(bpm_value)
+            bpm_value.text = bytes.decode(i["bpm"])
+            tp.append(bpm)
+        # Position
+        if not bytes.decode(i["lati"]) == "None":
+            position = ET.Element("Position")
+            tp.append(position)
+        #   LatitudeDegrees
+            lati = ET.Element("LatitudeDegrees")
+            lati.text = bytes.decode(i["lati"])
+            position.append(lati)
+        #   LongitudeDegrees
+            longi = ET.Element("LongitudeDegrees")
+            longi.text = bytes.decode(i["longi"])
+            position.append(longi)
+    
+    # write to TCX file
+    tree.write("../TCX_OUT/" + fit_id + ".tcx", encoding='utf-8', xml_declaration=True)
```

Comment:
use `GPX_FOLDER` instead

Reply:
do i need to export tcx files to gpx folder? and where is that constant please, i didn't find it

Reply:
https://github.com/yihong0618/running_page/blob/3ae13b477ef75034ff455ae1fe3163f041afb610/scripts/config.py#L11
Sorry for my typo, it's `TCX_FOLDER`

Reply:
got it thx